vyper/semantics/types/indexable/mapping.py

```diff
@@ -40,6 +40,10 @@ def from_annotation(
         is_public: bool = False,
         is_immutable: bool = False,
     ) -> MappingDefinition:
+        if isinstance(node, vy_ast.Subscript) and isinstance(
```

Comment:
i think instead of this ad-hoc check (which checks for a specific user pattern), we should just improve the error message in the other branch.